stdlib/public/core/OutputStream.swift

```diff
@@ -72,11 +72,18 @@ public protocol TextOutputStream {
 
   /// Appends the given string to the stream.
   mutating func write(_ string: String)
+
+  mutating func _writeASCII(_ buffer: UnsafeBufferPointer<UInt8>)
 }
 
 extension TextOutputStream {
   public mutating func _lock() {}
   public mutating func _unlock() {}
+
+  @inlinable
+  public mutating func _writeASCII(_ buffer: UnsafeBufferPointer<UInt8>) {
```

Comment:
Err crap, is it because we can't have non-public requirements on public protocols?

BTW, I think that floating point types' `TextOutputStreamable.write<Target>` should have a `@_specialize(where Target == String` for the 99% case, which is also the perf-sensitive case. That should give us better perf, and since that's the only call site for this, we can form an immortal String there.

Longer-term, we don't really care for a buffer-returning function for `_float${Bits}ToString`, we just want to write directly into the allocated extra capacity. Same for Int->String, etc. CC @stephentyrone who might be looking at that.

Reply:
I agree that it could make sense to specialize `write(to:)` for `String` and possibly `DefaultStringInterpolation`.

The implementation and use of `_float${Bits}ToString` is entirely behind resilience barriers, so we can change it in the future. I suppose what we can't do in the future, though, is replace `_writeASCII(_:)` with whatever primitives we'd need to have a stream allocate some empty space, let us write into it, and then tell it how much of that space we used.